sequencer/src/hotshot_commitment.rs

```diff
@@ -132,20 +143,37 @@ impl HotShotDataSource for HotShotClient {
     }
 }
 
+#[derive(Debug)]
+enum SyncError {
+    TransactionFailed { err: anyhow::Error, num_leaves: u64 },
+    Other(anyhow::Error),
+}
+
 async fn sync_with_l1(
     max_blocks: u64,
     hotshot: &impl HotShotDataSource,
     contract: &HotShot<Signer>,
-) -> Result<(), anyhow::Error> {
-    let contract_block_height = contract.block_height().call().await?.as_u64();
+) -> Result<(), SyncError> {
+    let contract_block_height = contract
+        .block_height()
+        .call()
+        .await
+        .map_err(|e| SyncError::Other(e.into()))?
```

Comment:
Nit: implement `From<anyhow::Error> for SyncError` and you can just use `?` without this `map_err`

Reply:
But this underlying error isn't an anyhow:Error, so we'd still need to do a double mapping somehow right?